Cleanup gateway vhost config gen

frankbu opened the pull request:
Fixes #12655

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/12847#" title="Author self-approved">frankbu</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **costinm**

If they are not already assigned, you can assign the PR to them by writing `/assign @costinm` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[pilot/OWNERS](https://github.com/istio/istio/blob/release-1.1/pilot/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["costinm"]} -->

istio-testing commented:
@frankbu: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-integ-k8s-tests.sh | 22d5e3615689edd290b713f52223a0616b1ff6de | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-integ-k8s-tests/5679) | `/test istio-integ-k8s-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@frankbu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-integ-k8s-tests.sh | 22d5e3615689edd290b713f52223a0616b1ff6de | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-integ-k8s-tests/5679) | `/test istio-integ-k8s-tests`
prow/istio-pilot-multicluster-e2e.sh | 22d5e3615689edd290b713f52223a0616b1ff6de | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-multicluster-e2e/10095) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

rshriram commented on pilot/pkg/model/service.go:
```diff
@@ -639,6 +639,55 @@ func (h Hostnames) Swap(i, j int) {
+		}
+	}
+	return result
+}
```
could you add a test for this? that will be the ultimate validation and prevent people from messing this logic up

rshriram commented on pilot/pkg/model/service.go:
```diff
@@ -639,6 +639,55 @@ func (h Hostnames) Swap(i, j int) {
+// HostnamesForNamespace returns the subset of hosts that are in the specified namespace.
+// The list of hosts contains host names optionally qualified with namespace/.
+// If not qualified, a host name is considered to be in any namespace.
+// NOTE: only expecting ns/host and no ./* or */host.
```
server host could definitely be of the form */host

rshriram commented on pilot/pkg/model/service.go:
```diff
@@ -639,6 +639,55 @@ func (h Hostnames) Swap(i, j int) {
+//  Hostnames(["foo.com","*.net"]).Intersection(Hostnames(["*.com","bar.net"])) = Hostnames(["foo.com","bar.net"])
+//  Hostnames(["foo.com","*.net"]).Intersection(Hostnames(["*.bar.net"]))       = Hostnames(["*.bar.net"])
+//  Hostnames(["foo.com"]).Intersection(Hostnames(["bar.com"]))                 = Hostnames([])
+func (h Hostnames) Intersection(other Hostnames) Hostnames {
```
need additional cases here. `Hostnames([]).Intersection(Hostnames(["bar.com"]) = Hostnames([])` and vice versa

rshriram commented on pilot/pkg/networking/core/v1alpha3/gateway.go:
```diff
@@ -216,71 +216,58 @@ func (configgen *ConfigGeneratorImpl) buildGatewayHTTPRouteConfig(env *model.Env
+			// 2. virtualService hosts are *.foo.com, and server hosts are ns/1.foo.com, ns/2.foo.com, ns/3.foo.com
+			intersectingHosts := serverHosts.Intersection(virtualServiceHosts)
+			if len(intersectingHosts) == 0 {
+				log.Debugf("%s virtual service %q has no matching hosts for gateways %v server %d", node.ID, virtualService.Name, gatewayName, port)
```
I would remove this.. this will be super noisy when debugging pilot

istio-testing commented:
@frankbu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | 22d5e3615689edd290b713f52223a0616b1ff6de | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-multicluster-e2e/10095) | `/test istio-pilot-multicluster-e2e`
prow/istio-integ-k8s-tests.sh | a94471fe11ddf1885cedfabfec0d32b6a757bfa5 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-integ-k8s-tests/5774) | `/test istio-integ-k8s-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@frankbu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-integ-k8s-tests.sh | a94471fe11ddf1885cedfabfec0d32b6a757bfa5 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-integ-k8s-tests/5774) | `/test istio-integ-k8s-tests`
prow/istio-pilot-multicluster-e2e.sh | a94471fe11ddf1885cedfabfec0d32b6a757bfa5 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-multicluster-e2e/10185) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@frankbu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | a94471fe11ddf1885cedfabfec0d32b6a757bfa5 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-multicluster-e2e/10185) | `/test istio-pilot-multicluster-e2e`
prow/istio-integ-k8s-tests.sh | 7865c22488ed9ba6c1d41f40642c15003b9cc8c3 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-integ-k8s-tests/5784) | `/test istio-integ-k8s-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@frankbu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-integ-k8s-tests.sh | 7865c22488ed9ba6c1d41f40642c15003b9cc8c3 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-integ-k8s-tests/5784) | `/test istio-integ-k8s-tests`
prow/istio-pilot-multicluster-e2e.sh | 7865c22488ed9ba6c1d41f40642c15003b9cc8c3 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-multicluster-e2e/10195) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@frankbu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | 7865c22488ed9ba6c1d41f40642c15003b9cc8c3 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-multicluster-e2e/10195) | `/test istio-pilot-multicluster-e2e`
prow/istio-integ-k8s-tests.sh | 53ab249d8e665027b992c34996b6d56a89abc05c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-integ-k8s-tests/5818) | `/test istio-integ-k8s-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@frankbu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-integ-k8s-tests.sh | 53ab249d8e665027b992c34996b6d56a89abc05c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-integ-k8s-tests/5818) | `/test istio-integ-k8s-tests`
prow/istio-pilot-multicluster-e2e.sh | 53ab249d8e665027b992c34996b6d56a89abc05c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-multicluster-e2e/10225) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@frankbu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | 53ab249d8e665027b992c34996b6d56a89abc05c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-multicluster-e2e/10225) | `/test istio-pilot-multicluster-e2e`
prow/istio-integ-k8s-tests.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-integ-k8s-tests/5878) | `/test istio-integ-k8s-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@frankbu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | 53ab249d8e665027b992c34996b6d56a89abc05c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-multicluster-e2e/10225) | `/test istio-pilot-multicluster-e2e`
prow/istio-integ-k8s-tests.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-integ-k8s-tests/5878) | `/test istio-integ-k8s-tests`
prow/e2e-simpleTests.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-simpleTests/19577) | `/test e2e-simpleTests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@frankbu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | 53ab249d8e665027b992c34996b6d56a89abc05c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-multicluster-e2e/10225) | `/test istio-pilot-multicluster-e2e`
prow/istio-integ-k8s-tests.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-integ-k8s-tests/5878) | `/test istio-integ-k8s-tests`
prow/e2e-simpleTests.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-simpleTests/19577) | `/test e2e-simpleTests`
prow/e2e-simpleTests-minProfile.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-simpleTestsMinProfile/3970) | `/test e2e-simpleTestsMinProfile`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@frankbu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | 53ab249d8e665027b992c34996b6d56a89abc05c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-multicluster-e2e/10225) | `/test istio-pilot-multicluster-e2e`
prow/istio-integ-k8s-tests.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-integ-k8s-tests/5878) | `/test istio-integ-k8s-tests`
prow/e2e-simpleTests.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-simpleTests/19577) | `/test e2e-simpleTests`
prow/e2e-simpleTests-minProfile.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-simpleTestsMinProfile/3970) | `/test e2e-simpleTestsMinProfile`
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-e2e-envoyv2-v1alpha3/17516) | `/test istio-pilot-e2e-envoyv2-v1alpha3`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

frankbu commented:
@rshriram, I addressed your comments and added some unit tests. I also tested the vhost generation code by manually running the below config. It did turn up a couple of bugs, one with the Gateway host validator, and one with the intersection function returning "foo.com" twice, since it matched two of the gateway (overlapping) hosts. The result added the "foo.com" routes twice in the vhost, which would work but not so nice.

We should probably add the following test, or something similar, to the build tests, but I think that can be left for a future PR. PTAL.

```
# This config should produce the folloiwing vhosts in ingressgateway: 
#
#  "blah.com:80"
#    "/blahNs2path"
#
#  "foo.com:80"
#    "/fooNs1path"
#    "/fooNs2path"
#    "/fooNs3path"
#    "/starNs4path"
#
#  "h1.host.com:80"
#    "/h1ns1path2"
#    "/h1ns1path1"
#
#  "h2.host.com:80"
#    "/h2ns1path"
#
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  name: test-gw
  namespace: default
spec:
  selector:
    istio: ingressgateway
  servers:
  - hosts:
    - "ns1/*"
    - "*/foo.com"
    - "ns2/blah.com"
    port:
      name: http
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: v1
  namespace: ns1
spec:
  gateways:
  - default/test-gw
  hosts:
  - h1.host.com
  http:
  - match:
    - uri:
        exact: /h1ns1path1
    route:
    - destination:
        host: productpage
        port:
          number: 9080
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: v2
  namespace: ns1
spec:
  gateways:
  - default/test-gw
  hosts:
  - h1.host.com
  http:
  - match:
    - uri:
        exact: /h1ns1path2
    route:
    - destination:
        host: productpage
        port:
          number: 9080
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: v3
  namespace: ns1
spec:
  gateways:
  - default/test-gw
  hosts:
  - h2.host.com
  http:
  - match:
    - uri:
        exact: /h2ns1path
    route:
    - destination:
        host: productpage
        port:
          number: 9080
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: v4
  namespace: ns1
spec:
  gateways:
  - default/test-gw
  hosts:
  - foo.com
  http:
  - match:
    - uri:
        exact: /fooNs1path
    route:
    - destination:
        host: productpage
        port:
          number: 9080
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: v5
  namespace: ns2
spec:
  gateways:
  - default/test-gw
  hosts:
  - foo.com
  http:
  - match:
    - uri:
        exact: /fooNs2path
    route:
    - destination:
        host: productpage
        port:
          number: 9080
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: v6
  namespace: ns2
spec:
  gateways:
  - default/test-gw
  hosts:
  - h1.host.com
  http:
  - match:
    - uri:
        exact: /h1ns2path
    route:
    - destination:
        host: productpage
        port:
          number: 9080
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: v7
  namespace: ns2
spec:
  gateways:
  - default/test-gw
  hosts:
  - blah.com
  http:
  - match:
    - uri:
        exact: /blahNs2path
    route:
    - destination:
        host: productpage
        port:
          number: 9080
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: v8
  namespace: ns3
spec:
  gateways:
  - default/test-gw
  hosts:
  - h1.host.com
  http:
  - match:
    - uri:
        exact: /h1ns3path
    route:
    - destination:
        host: productpage
        port:
          number: 9080
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: v9
  namespace: ns3
spec:
  gateways:
  - default/test-gw
  hosts:
  - blah.com
  http:
  - match:
    - uri:
        exact: /blahNs3path
    route:
    - destination:
        host: productpage
        port:
          number: 9080
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: v10
  namespace: ns3
spec:
  gateways:
  - default/test-gw
  hosts:
  - foo.com
  http:
  - match:
    - uri:
        exact: /fooNs3path
    route:
    - destination:
        host: productpage
        port:
          number: 9080
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: v11
  namespace: ns4
spec:
  gateways:
  - default/test-gw
  hosts:
  - "*"
  http:
  - match:
    - uri:
        exact: /starNs4path
    route:
    - destination:
        host: productpage
        port:
          number: 9080
```

istio-testing commented:
@frankbu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | 53ab249d8e665027b992c34996b6d56a89abc05c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-multicluster-e2e/10225) | `/test istio-pilot-multicluster-e2e`
prow/istio-integ-k8s-tests.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-integ-k8s-tests/5878) | `/test istio-integ-k8s-tests`
prow/e2e-simpleTests.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-simpleTests/19577) | `/test e2e-simpleTests`
prow/e2e-simpleTests-minProfile.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-simpleTestsMinProfile/3970) | `/test e2e-simpleTestsMinProfile`
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-e2e-envoyv2-v1alpha3/17516) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/e2e-dashboard.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-dashboard/14216) | `/test e2e-dashboard`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@frankbu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | 53ab249d8e665027b992c34996b6d56a89abc05c | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-multicluster-e2e/10225) | `/test istio-pilot-multicluster-e2e`
prow/istio-integ-k8s-tests.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-integ-k8s-tests/5878) | `/test istio-integ-k8s-tests`
prow/e2e-simpleTests.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-simpleTests/19577) | `/test e2e-simpleTests`
prow/e2e-simpleTests-minProfile.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-simpleTestsMinProfile/3970) | `/test e2e-simpleTestsMinProfile`
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-e2e-envoyv2-v1alpha3/17516) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/e2e-dashboard.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-dashboard/14216) | `/test e2e-dashboard`
prow/e2e-bookInfoTests-v1alpha3.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-bookInfoTests-envoyv2-v1alpha3/16631) | `/test e2e-bookInfoTests-envoyv2-v1alpha3`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@frankbu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-integ-k8s-tests.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-integ-k8s-tests/5878) | `/test istio-integ-k8s-tests`
prow/e2e-simpleTests.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-simpleTests/19577) | `/test e2e-simpleTests`
prow/e2e-simpleTests-minProfile.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-simpleTestsMinProfile/3970) | `/test e2e-simpleTestsMinProfile`
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-e2e-envoyv2-v1alpha3/17516) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/e2e-dashboard.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-dashboard/14216) | `/test e2e-dashboard`
prow/e2e-bookInfoTests-v1alpha3.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-bookInfoTests-envoyv2-v1alpha3/16631) | `/test e2e-bookInfoTests-envoyv2-v1alpha3`
prow/istio-pilot-multicluster-e2e.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-multicluster-e2e/10284) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@frankbu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-integ-k8s-tests.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-integ-k8s-tests/5878) | `/test istio-integ-k8s-tests`
prow/e2e-simpleTests.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-simpleTests/19577) | `/test e2e-simpleTests`
prow/e2e-simpleTests-minProfile.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-simpleTestsMinProfile/3970) | `/test e2e-simpleTestsMinProfile`
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-e2e-envoyv2-v1alpha3/17516) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/e2e-dashboard.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-dashboard/14216) | `/test e2e-dashboard`
prow/e2e-bookInfoTests-v1alpha3.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-bookInfoTests-envoyv2-v1alpha3/16631) | `/test e2e-bookInfoTests-envoyv2-v1alpha3`
prow/istio-pilot-multicluster-e2e.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-multicluster-e2e/10284) | `/test istio-pilot-multicluster-e2e`
prow/e2e-mixer-no_auth.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-mixer-no_auth/14486) | `/test e2e-mixer-no_auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@frankbu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-integ-k8s-tests.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-integ-k8s-tests/5878) | `/test istio-integ-k8s-tests`
prow/e2e-simpleTests.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-simpleTests/19577) | `/test e2e-simpleTests`
prow/e2e-simpleTests-minProfile.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-simpleTestsMinProfile/3970) | `/test e2e-simpleTestsMinProfile`
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-e2e-envoyv2-v1alpha3/17516) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/e2e-dashboard.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-dashboard/14216) | `/test e2e-dashboard`
prow/e2e-bookInfoTests-v1alpha3.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-bookInfoTests-envoyv2-v1alpha3/16631) | `/test e2e-bookInfoTests-envoyv2-v1alpha3`
prow/istio-pilot-multicluster-e2e.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-multicluster-e2e/10284) | `/test istio-pilot-multicluster-e2e`
prow/e2e-mixer-no_auth.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-mixer-no_auth/14486) | `/test e2e-mixer-no_auth`
prow/e2e-bookInfoTests-trustdomain.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-bookInfoTests-trustdomain/119) | `/test e2e-bookInfoTests-trustdomain`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@frankbu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-integ-k8s-tests.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-integ-k8s-tests/5878) | `/test istio-integ-k8s-tests`
prow/e2e-simpleTests.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-simpleTests/19577) | `/test e2e-simpleTests`
prow/e2e-simpleTests-minProfile.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-simpleTestsMinProfile/3970) | `/test e2e-simpleTestsMinProfile`
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-e2e-envoyv2-v1alpha3/17516) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/e2e-dashboard.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-dashboard/14216) | `/test e2e-dashboard`
prow/e2e-bookInfoTests-v1alpha3.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-bookInfoTests-envoyv2-v1alpha3/16631) | `/test e2e-bookInfoTests-envoyv2-v1alpha3`
prow/istio-pilot-multicluster-e2e.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-multicluster-e2e/10284) | `/test istio-pilot-multicluster-e2e`
prow/e2e-mixer-no_auth.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-mixer-no_auth/14486) | `/test e2e-mixer-no_auth`
prow/e2e-bookInfoTests-trustdomain.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-bookInfoTests-trustdomain/119) | `/test e2e-bookInfoTests-trustdomain`
prow/e2e_pilotv2_auth_sds.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio_auth_sds_e2e/3845) | `/test istio_auth_sds_e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

frankbu commented:
/retest

rshriram commented on pilot/pkg/model/validation.go:
```diff
@@ -446,10 +446,6 @@ func validateServer(server *networking.Server) (errs error) {
-			// short name hosts are not allowed in gateways
-			if host != "*" && !strings.Contains(host, ".") {
-				errs = appendErrors(errs, fmt.Errorf("short names (non FQDN) are not allowed in Gateway server hosts"))
-			}
```
why remove this?

frankbu replied:
This check fails for a host like  `ns1/*`, which is valid. The correct check for short hosts is done in `validateNamespaceSlashWildcardHostname`, just below.

rshriram commented:
/retest

frankbu commented:
/test e2e-simpleTestsMinProfile

istio-testing commented:
@frankbu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-e2e-envoyv2-v1alpha3/17516) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/e2e-bookInfoTests-v1alpha3.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-bookInfoTests-envoyv2-v1alpha3/16631) | `/test e2e-bookInfoTests-envoyv2-v1alpha3`
prow/istio-pilot-multicluster-e2e.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio-pilot-multicluster-e2e/10284) | `/test istio-pilot-multicluster-e2e`
prow/e2e-mixer-no_auth.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-mixer-no_auth/14486) | `/test e2e-mixer-no_auth`
prow/e2e-bookInfoTests-trustdomain.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/e2e-bookInfoTests-trustdomain/119) | `/test e2e-bookInfoTests-trustdomain`
prow/e2e_pilotv2_auth_sds.sh | 80122be52126a6ddeaae52f89c20210ff82e4f1b | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12847/istio_auth_sds_e2e/4008) | `/test istio_auth_sds_e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->
